### PR TITLE
Updated list display of demo pods

### PR DESCRIPTION
**Pull request by samhotep**
## Done

- Switched to cards instead of using a table
- Added search 
- Added status labels to jobs
- Added start/stop buttons to each pod
- Added filters for status

## QA

- Check the demo is running at https://demos-haus-68.demos.haus/
- Test that search is working
- Test that filters are working (independently of search)
- Test that clicking start/stop actually changes the state of a deployment (try starting a failed build e.g. lxd-ui-307-demos-haus)

## Issue / Card

Fixes [WD-7572](https://warthogs.atlassian.net/browse/WD-7572?atlOrigin=eyJpIjoiZTFhZTJjZDBlNDJiNGNlZGExODU1MzU4MGVjNDkwMmYiLCJwIjoiaiJ9)

[WD-7572]: https://warthogs.atlassian.net/browse/WD-7572?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jw-can** [comment]
@samhotep Neither start nor stop work for me... (using demo)

**Comment by samhotep** [comment]
> @samhotep Neither start nor stop work for me... (using demo)

There was a k8s auth issue. It should be working now

**Comment by jw-can** [comment]
@samhotep I see some issues with reported status:
- lxd-ui-681-demos-haus is shown as "Building" but is actually running
- when stopping a demo it remains as running

**Comment by samhotep** [comment]
> @samhotep I see some issues with reported status:
> 
> * lxd-ui-681-demos-haus is shown as "Building" but is actually running
> * when stopping a demo it remains as running

Looking into it!

**Comment by samhotep** [comment]
@jw-can I've made some updates, please take a look

**Review by jw-can — DISMISSED** [review]
There's a strange behaviour for the search/filter dynamic. If I enter "7" I get results, when I add a filter for "Failed" there's no results. If I add another "1" (so "71") there'll be results again, even with activated "Failed" filter.

Maybe you could also always display the filters, so that one can directly find demos that failed building?

**Comment by samhotep** [comment]
> There's a strange behaviour for the search/filter dynamic. If I enter "7" I get results, when I add a filter for "Failed" there's no results. If I add another "1" (so "71") there'll be results again, even with activated "Failed" filter.
> 
> Maybe you could also always display the filters, so that one can directly find demos that failed building?

That's a good catch. Looks like the input overwrites whatever is in the buffer, so the previous search is lost if you switch between using typing and the filters. Fix incoming!

Also, will show the filters by default.

**Comment by petesfrench** [comment]
Am I supposed to be able to filter without typing anything into the search field? 
image

File: static/js/src/demolist.tsx [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        }}
+      >
+        <div
+          style={{
+            display: "flex",
+            alignItems: "center",
+            justifyContent: "center",
+          }}
+        >
+          <a>
```
[the remark this answers is not in the document]
Comment (samhotep) [reply static/js/src/demolist.tsx:116]: Yes, yes!

**Comment by samhotep** [comment]
> Am I supposed to be able to filter without typing anything into the search field? image

Oops, it should be open. Fix incoming

**Comment by samhotep** [comment]
@petesfrench I've made the filter independent of the search for now

**Review by petesfrench — APPROVED** [review]
LGTM

File: templates/index.html [hunk 2]
```diff
@@ -7,31 +7,64 @@
...
-    <link rel="preconnect" href="https://assets.ubuntu.com">
+    <link rel="preconnect" href="https://assets.ubuntu.com" />
     <!-- <link rel="dns-prefetch" href="http://www.spreadfirefox.com/"> -->
-    
+
     <!-- Add inline JS below -->
-    <script>// Inline JS</script>
-    
+    <script>
+      // Inline JS
```
[the remark this answers is not in the document]
Comment (samhotep) [reply templates/index.html:20]: I've updated the filter to be independent of the search

**Pull request merged**